middleware/metrics_test.go

```diff
@@ -0,0 +1,138 @@
+package middleware
+
+import (
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var tR = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "http_requests_total",
+		Help: "Number of get requests.",
+	},
+	[]string{"path"},
+)
+
+var rS = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "response_status",
+		Help: "Status of HTTP response",
+	},
+	[]string{"status"},
+)
+
+var hD = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Name: "http_response_time_seconds",
+	Help: "Duration of HTTP requests.",
+}, []string{"path"})
+
+func TestPrometheusRegisters(t *testing.T) {
+	assert := assert.New(t)
+
+	t.Log("Registering Prometheus Metrics")
+
+	prometheus.Register(tR)
+	prometheus.Register(responseStatus)
+	prometheus.Register(httpDuration)
```

Comment:
Still not against Init(), basically just need to decide if we're okay with calling prometheus registrations when we import the package, and if so, I'll move them there. It actually massively simplifies everything. Although since these are supposed to be specially constructed for this test, would still need an init here. :P